qemu_v8.mk

```diff
@@ -41,15 +47,62 @@ QEMU_PATH		?= $(ROOT)/qemu
 QEMU_BUILD		?= $(QEMU_PATH)/build
 SOC_TERM_PATH		?= $(ROOT)/soc_term
 MODULE_OUTPUT		?= $(ROOT)/out/kernel_modules
+UBOOT_PATH		?= $(ROOT)/u-boot
+UBOOT_BIN		?= $(UBOOT_PATH)/u-boot.bin
+MKIMAGE_PATH		?= $(UBOOT_PATH)/tools
+
+ROOTFS_GZ		?= $(BINARIES_PATH)/rootfs.cpio.gz
+ROOTFS_UGZ		?= $(BINARIES_PATH)/rootfs.cpio.uboot
+
+KERNEL_IMAGE		?= $(LINUX_PATH)/arch/arm64/boot/Image
+KERNEL_IMAGEGZ		?= $(LINUX_PATH)/arch/arm64/boot/Image.gz
+KERNEL_UIMAGE		?= $(BINARIES_PATH)/uImage
+
+# Load and entry addresses
+KERNEL_ENTRY		?= 0x40400000
+KERNEL_LOADADDR		?= 0x40400000
+ROOTFS_ENTRY		?= 0x44000000
+ROOTFS_LOADADDR		?= 0x44000000
+
+ifeq ($(GICV3),y)
+	TFA_GIC_DRIVER	?= QEMU_GICV3
+	QEMU_GIC_VERSION = 3
+else
+	TFA_GIC_DRIVER	?= QEMU_GICV2
+	QEMU_GIC_VERSION = 2
+endif
 
+ifeq ($(UBOOT),y)
+BL33_BIN		?= $(UBOOT_BIN)
+BL33_DEPS		?= uboot
+else
+BL33_BIN		?= $(EDK2_BIN)
+BL33_DEPS		?= edk2
+endif
 
 ################################################################################
 # Targets
 ################################################################################
-all: arm-tf buildroot edk2 linux optee-os qemu soc-term
-clean: arm-tf-clean buildroot-clean edk2-clean linux-clean optee-os-clean \
+TARGET_DEPS ?= arm-tf buildroot linux optee-os qemu soc-term
```

Comment:
Does it really make sense to have this overridable?

Overall I think that most of the `?=` locations in these make files should have `:=` or `=` instead. There are only few variables which it makes sense to override. Fixing this is out of scope for this PR, but there's no need to make matters worse.

Reply:
I will change TARGET_DEPS ?= to TARGET_DEPS :=